[bug]  java.lang.ArrayIndexOutOfBoundsException with user defined replicatorPrefix.

**Describe the bug**
The following exception occurs when I changed replicatorPrefix, and the replicator does not work.

Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 2 out of bounds for length 2
	at org.apache.pulsar.broker.service.Producer.<init>(Producer.java:131) ~[classes/:?]
	at org.apache.pulsar.broker.service.ServerCnx.buildProducerAndAddTopic(ServerCnx.java:1296) ~[classes/:?]
	at org.apache.pulsar.broker.service.ServerCnx.lambda$handleProducer$18(ServerCnx.java:1240) ~[classes/:?]

**To Reproduce**
Steps to reproduce the behavior:
1. set replicatorPrefix="my-replicatorPrefix" in config
2. start replicator
3. send message
4. Error occurs

**Expected behavior**
Replicator should works with user defined replicatorPrefix.

**Screenshots**
NO

**Desktop (please complete the following information):**
 - OS: macOS

**Additional context**
No

--- a/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Producer.java
+++ b/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Producer.java
@@ -126,9 +126,10 @@ public class Producer {
         stats.metadata = this.metadata;
         stats.accessMode = Commands.convertProducerAccessMode(accessMode);
 
-        this.isRemote = producerName
-                .startsWith(cnx.getBrokerService().pulsar().getConfiguration().getReplicatorPrefix());
-        this.remoteCluster = parseRemoteClusterName(producerName, isRemote);
+
+        String replicatorPrefix = cnx.getBrokerService().pulsar().getConfiguration().getReplicatorPrefix() + ".";
+        this.isRemote = producerName.startsWith(replicatorPrefix);
+        this.remoteCluster = parseRemoteClusterName(producerName, isRemote, replicatorPrefix);
 
         this.isEncrypted = isEncrypted;
         this.schemaVersion = schemaVersion;
@@ -138,9 +139,14 @@ public class Producer {
         this.clientAddress = cnx.clientSourceAddress();
     }
 
-    private String parseRemoteClusterName(String producerName, boolean isRemote) {
+    /**
+     * Producer name for replicator is in format:
+     * "replicatorPrefix.localCluster" (old)
+     * "replicatorPrefix.localCluster-->remoteCluster" (new)
+     */
+    private String parseRemoteClusterName(String producerName, boolean isRemote, String replicatorPrefix) {
         if (isRemote) {
-            String clusterName = producerName.split("\\.")[2];
+            String clusterName = producerName.substring(replicatorPrefix.length());
             return clusterName.contains(REPL_PRODUCER_NAME_DELIMITER)
                     ? clusterName.split(REPL_PRODUCER_NAME_DELIMITER)[0] : clusterName;
         }

--- a/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Producer.java
+++ b/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Producer.java
@@ -140,7 +140,7 @@ public class Producer {
     }
 
     /**
-     * Producer name for replicator is in format:
+     * Producer name for replicator is in format.
      * "replicatorPrefix.localCluster" (old)
      * "replicatorPrefix.localCluster-->remoteCluster" (new)
      */

--- a/pulsar-broker/src/test/java/org/apache/pulsar/broker/service/ReplicatorTest.java
+++ b/pulsar-broker/src/test/java/org/apache/pulsar/broker/service/ReplicatorTest.java
@@ -730,6 +730,32 @@ public class ReplicatorTest extends ReplicatorTestBase {
         assertTrue(remoteClusters.contains("r1"));
     }
 
+    @Test(priority = 5, timeOut = 30000)
+    public void testReplicatorProducerNameWithUserDefinedReplicatorPrefix() throws Exception {
+        log.info("--- Starting ReplicatorTest::testReplicatorProducerNameWithUserDefinedReplicatorPrefix ---");
+        final String topicName = BrokerTestUtil.newUniqueName(
+                "persistent://pulsar/ns/testReplicatorProducerNameWithUserDefinedReplicatorPrefix");
+        final TopicName dest = TopicName.get(topicName);
+
+        pulsar1.getConfiguration().setReplicatorPrefix("user-defined-prefix");
+        pulsar2.getConfiguration().setReplicatorPrefix("user-defined-prefix");
+        pulsar3.getConfiguration().setReplicatorPrefix("user-defined-prefix");
+
+        @Cleanup
+        MessageProducer producer1 = new MessageProducer(url1, dest);
+
+        Awaitility.await().untilAsserted(()->{
+            assertTrue(pulsar2.getBrokerService().getTopicReference(topicName).isPresent());
+        });
+        Optional<Topic> topic = pulsar2.getBrokerService().getTopicReference(topicName);
+        assertTrue(topic.isPresent());
+        Set<String> remoteClusters = topic.get().getProducers().values().stream()
+                .map(org.apache.pulsar.broker.service.Producer::getRemoteCluster)
+                .collect(Collectors.toSet());
+        assertTrue(remoteClusters.contains("r1"));
+    }
+
+
     /**
      * Issue #199
      *

